### PR TITLE
Skip Client layer without buffer

### PR DESCRIPTION
**Pull request by Shao-Feng**
The Client layer composed by SF could be sent to HWC.
and trigger a error log below
HWC don't support layer without buffer if not in type SolidColor

Change-Id: I76bb79520453c68a0b17fbf69db0a151d993bbb4
Tests: Work on Android Q
Tracked-On: None
Signed-off-by: Shaofeng Tang <shaofeng.tang@intel.com>

### CODE REVIEW

File: os/android/iahwc2.cpp [hunk 1]
```diff
@@ -637,6 +637,14 @@ HWC2::Error IAHWC2::HwcDisplay::PresentDisplay(int32_t *retire_fence) {
       cursor_z_order = l.second.z_order();
       continue;
     }
+#ifndef FORCE_ALL_DEVICE_TYPE
+    if ((l.second.validated_type() == HWC2::Composition::Client) &&
```
Comment (YuanjunHuang) [inline os/android/iahwc2.cpp]: do you know in what case and why this situation will be met?

Comment (js0701) [reply os/android/iahwc2.cpp]: it is a MACRO added by Dong. Default should be force

Comment (js0701) [reply os/android/iahwc2.cpp]: @dyang23  Please also help to review

Comment (Shao-Feng) [reply os/android/iahwc2.cpp]: It is SurfaceFlinger's behavior. Now, we can meet it in Home and App page. I suppose when SF do composition for Client layer, this issue could happen.

Comment (Shao-Feng) [reply os/android/iahwc2.cpp]: Till now, this MACRO is not defined. it is not force all layer "Device". I will prepare one patch for it

**Comment by dyang23** [comment]
LGTM

**Pull request merged**